Making german translation of the docs easier to read.

frommMoritz opened the pull request:
The German translation was, and in parts still is, not so easy to read since it sometimes uses incorrect scentense forms. Spelling may not be 100% correct, but it should be less "angular" to read.

frommMoritz committed:
Making german translation of the docs easier to read.

The German translation was, and in parts still is, not so easy to read since it sometimes uses incorrect scentense forms. Spelling may not be 100% correct, but it should be less "angular" to read.

frommMoritz committed:
Updating last Change

Maikuolan commented:
Cheers.  :+1: 😊

Because both #95 and #96 include work done on the same file, when I accept one of them, it will probably create merge conflicts for the other, I think. Maybe we should combine the changes into a single PR (and then close the other), so that we can include all the changes together? Will await your thoughts on this. 😊

Maikuolan commented:
Also, to clarify:

Where `2 und 5` was changed to `3 und 5` (under the heading `2.1 INSTALLATION MIT COMPOSER`), was that intentional or by mistake? Either way, no problem. Good to clarify though, in case I've missed something myself (and if it's a mistake, I can easily change it back later anyway).

I notice also that there's a line in there which has been `<!-- commented out -->` - Is that needed, or can that be removed?

DanielRuf reviewed (commented):
I'll have to check the changes.

frommMoritz commented:
Thanks for your feedback @Maikuolan, 

> Because both #95 and #96 include work done on the same file, when I accept one of them, it will probably create merge conflicts for the other, I think. Maybe we should combine the changes into a single PR (and then close the other), so that we can include all the changes together? Will await your thoughts on this. blush

If this is ok for you, I just could copy the changes made to the german file in #95 and add them to this PR. But to keep everything nice and organized I still would leave #95 open and reduce it to just the changes in the English file?

> Where 2 und 5 was changed to 3 und 5 (under the heading 2.1 INSTALLATION MIT COMPOSER), was that intentional or by mistake? Either way, no problem. Good to clarify though, in case I've missed something myself (and if it's a mistake, I can easily change it back later anyway).

Yes, totally my fault. I've checked back with the English version and the guide to manual installation, it seems to me as if this may be a typo or something.

> I notice also that there's a line in there which has been <!-- commented out --> - Is that needed, or can that be removed?

No, that is not necessary. I'll remove it.

frommMoritz committed:
Implementing feedback from #96

Maikuolan commented:
This PR has been quiet for a while now. No rush at all, but I'm keen to add some new information to the README file soon (already done for other language versions; I've just held off on doing so for German for the moment, due to this PR being open currently).

Are there any uncommited changes which currently exist? Also, I'm wondering whether I should try committing directly to this branch, or to a separate branch, or ignore this PR and commit directly to master or whatever else.

I'll likely be out of town for the next two days at least, so there's no rush to reply either, but when you've got a moment and when convenient, let me know what you reckon. Cheers. @frommMoritz @DanielRuf

DanielRuf commented:
> Are there any uncommited changes which currently exist? Also, I'm wondering whether I should try committing directly to this branch, or to a separate branch, or ignore this PR and commit directly to master or whatever else.

Regarding which changes that you want to make?

Maikuolan commented:
> Regarding which changes that you want to make?

Mostly just wanting to sync the German language version with the changes made to the other language versions by 79822ba0ecbb9580c12da7d132601af4c4a3188a and 4d06e69b923a68da7c7f61e07573f230daa58025:
- Added an index/nav table (?) to the beginning of the configuration section, to make it easier to find specific entries in that section.
- Configuration entry about the `timezone` directive was missing (fixed).
- Added a small notice to the beginning of the `PHPMailer` category (specifically, that it can be ignored, if a user doesn't use 2FA).
- Added new documentation for the new rate limiting feature.
- Fixed the Composer installation instructions. The English version previously stated `you'll still need to prepare the configuration and hooks though; see "installing manually" steps 2 and 5`, which I've changed to `you'll still need to prepare the configuration, permissions and hooks though; see "installing manually" steps 2, 4, and 5`.

The previously missing information, and the newly added information, *probably* won't cause merge conflicts (at least, I think.. but my merge-speak isn't quite native-level, if that makes sense.. so, I think, but I'm not 100% sure). GitHub seems to be able to handle markdown files a little better than other types of files AFAICT, in regards to merging together unrelated branches of changes made to the same file, as long as those changes don't generally affect the same parts of the file. Though, the line I want to change about installing via Composer, would most likely cause a merge conflict, because it would involve modifying a line which has already been modified in commits made to this PR. So, I've refrained from touching the German language version for the moment, pending the conclusion of this PR.

Aside from that though.. Although my own German language skills are terrible, and I'm not personally in a position to judge corrections (though, I'm quite happy to trust a native or fluent speaker, for the same reason; both of you will certainly know better than I, as to the best way to phrase some particular thing in the German language version of the documentation, 😁), if we agree that the corrections are better, improve phrasing, etc, then I could probably help get the PR to the finish line a little faster by fixing some of the more obvious things, like missing full stops at the end of sentences, small typos in individual words, capitalisation, etc. But, if these things have already been done locally somewhere (i.e., as uncommited changes somewhere on the PR author's machine), then there's no point in my doing these changes again, and I'd be better off instead to just wait for them to be committed.

DanielRuf commented:
Do you mean a simple rebase?

DanielRuf commented:
> GitHub seems to be able to handle markdown files a little better than other types of files AFAICT, in regards to merging together unrelated branches of changes made to the 

No, it is done by Git (binary merge and so on) and if the same line hs changed in two different Git objects, it will produce a conflict.

Maikuolan commented:
> No, it is done by Git (binary merge and so on) and if the same line hs changed in two different Git objects, it will produce a conflict.

Ah, okay. Good that I've refrained then, instead of taking the risk.

DanielRuf commented:
I don't think that the author has prepared a rebase.

We should do this (locally, a rebase can be always aborted).

Plus: please KISS (keep it simple, stpupid and short) - the answers are often hard to read and very long =/.

DanielRuf commented:
> I don't think that the author has prepared a rebase.

I'll do a rebase later and fix any conflicts. On which branch can I rebase (master)?

Maikuolan commented:
Cheers.

DanielRuf commented:
https://github.com/CIDRAM/CIDRAM/compare/01c38f2c5fca28a87caf2d1a65406ffaadea1e8f..189fdfb0a9771f3b50a2612cbfb09e9eb3e1f5ca

Maikuolan committed:
Sync

Fixes a few typos and begins work to sync up with other language versions.

DanielRuf commented on _docs/readme.de.md:
```diff
@@ -1040,21 +1077,22 @@ Module wurden zur Verfügung gestellt, um sicherzustellen, dass die folgenden Pa
 - [Werden Probleme auftreten, wenn ich CIDRAM gleichzeitig mit CDNs oder Caching-Diensten verwende?](#CDN_CACHING_PROBLEMS)
 - [Wird CIDRAM meine Website vor DDoS-Angriffen schützen?](#DDOS_ATTACKS)
-- [Wenn ich Module oder Signaturdateien über die Update-Seite aktiviere oder deaktiviere, sortiert sie diese alphanumerisch in der Konfiguration. Kann ich die Art der Sortierung ändern?](#CHANGE_COMPONENT_SORT_ORDER)
+- [Wenn ich Module oder Signaturdateien über die Update-Seite aktiviere oder deaktiviere, sortiert siche diese alphanumerisch in der Konfiguration. Kann ich die Art der Sortierung ändern?](#CHANGE_COMPONENT_SORT_ORDER)
```
siche => sich

DanielRuf commented on _docs/readme.de.md:
```diff
@@ -543,10 +549,11 @@ Konfiguration der Signaturen.
 Wenn Sie möchten, können sie blockierten Benutzern mithilfe des google reCAPTCHA eine Möglichkeit bieten, durch den Beweis dass diese ein Mensch sind, auf die Seite zuzugreifen.
 
-Aufgrund der Risiken im Zusammenhang mit der Bereitstellung eines Wegs um die blokierung zu umgehen, ist es nicht empfohlen diese Methode bereitzustellen. Es ist Empfohlen, diese Methode nur zu aktivieren wenn Sie diese unbedingt benötigen. Dies könnten beispielsweise diese Situatuinen seien:
+Aufgrund der Risiken im Zusammenhang mit der Bereitstellung eines Wegs um die blockierung zu umgehen, ist es nicht empfohlen diese Methode bereitzustellen. Es ist Empfohlen, diese Methode nur zu aktivieren wenn Sie diese unbedingt benötigen. Dies könnten beispielsweise diese Situatuinen seien:
```
blockierung => Blockierung

DanielRuf commented on _docs/readme.de.md:
```diff
@@ -67,7 +67,7 @@ Oder in der `.htaccess` Datei:
 #### 2.1 INSTALLATION MIT COMPOSER
 
-Da [CIDRAM bei Packagist registriert ist](https://packagist.org/packages/cidram/cidram), können sie CIDRAM auch mittels composer installieren. Allerdings müssen sie immernoch die Hooks und die Konfiguration vorbereiten. (Siehe manuell installieren, Schritt 2 und 5)
+Da [CIDRAM bei Packagist registriert ist](https://packagist.org/packages/cidram/cidram), können sie CIDRAM auch mittels composer installieren. Allerdings müssen sie immernoch die Hooks und die Konfiguration vorbereiten. Siehe manuell installieren, Schritt 2, 4, und 5.
```
4, => 4

Maikuolan committed:
Progress + Implement review items.

DanielRuf commented on _docs/readme.de.md:
```diff
@@ -109,7 +109,7 @@ Das Front-End ist standardmäßig deaktiviert, um unautorisiert Zugriff zu verhi
 3) Loggen Sie sich mit dem standardmäßig Benutzernamen und Passwort ein (admin/password).
 
-Hinweis: **Um unautorisierten zugriff auf das Font-End zu verhindern, sollten sie sofort nach dem ersten Login ihren Benutzernamen und Passwort ändern. Dies ist notwendig, da mithilfe des Font-End möglich ist, beliebigen PhP Code auf ihre Webseite hochzuladen.**
+Hinweis: **Um unautorisierten zugriff auf das Front-End zu verhindern, sollten sie sofort nach dem ersten Login ihren Benutzernamen und Passwort ändern. Dies ist notwendig, da mithilfe des Front-End möglich ist, beliebigen PHP-Code auf ihre Webseite hochzuladen.**
```
Front-End => Frontend

ihre webseite => Ihre Webseite

DanielRuf commented on _docs/readme.de.md:
```diff
@@ -443,7 +446,7 @@ Wert | Produziert | Beschreibung
 
 ##### "log_banned_ips"
-- Sollen auch blockierte Anfragen von blockierten IP's Protokolliert werden? True = Ja [Standardeinstellung]; False = Nein.
+- Sollen auch blockierte Anfragen von verbotenen IPs Protokolliert werden? True = Ja [Standardeinstellung]; False = Nein.
```
Protokolliert => protokolliert

DanielRuf commented on _docs/readme.de.md:
```diff
@@ -514,7 +517,7 @@ Konfiguration der Signaturen.
 
 ##### "block_cloud"
-- CIDRs blockieren, welche zu Web-/Server- Hostern gehören. Wenn sie eine API betreiben oder erwarten, dass sie oft Aufrufe von solchen Servern bekommen sollten sie diese option auf false (nicht blockieren) setzen. Wenn sie solche Anfragen blockieren möchen, setzen sie diese Option auf true [Standarteinstellung]
+- CIDRs blockieren, welche zu Web-/Server- Hostern gehören. Wenn sie eine API betreiben oder erwarten, dass sie oft Aufrufe von solchen Servern bekommen sollten sie diese Option auf false (nicht blockieren) setzen. Wenn sie solche Anfragen blockieren möchen, setzen sie diese Option auf true [Standarteinstellung].
```
sie diese Option => Sie diese Option

DanielRuf replied:
sie eine API => Sie eine API
sie oft Aufrufe => Sie oft Aufrufe
sie solche Anfragen => Sie solche Anfragen
sie diese Option => Sie diese Option

"sie" is the second form of you (the formal you, not the personal you - like you greet an unknown person or an important person with respect)

Maikuolan replied:
Cheers. :+1:

Will fix now.

DanielRuf commented on _docs/readme.de.md:
```diff
@@ -549,7 +552,7 @@ Konfiguration der Signaturen.
 Wenn Sie möchten, können sie blockierten Benutzern mithilfe des google reCAPTCHA eine Möglichkeit bieten, durch den Beweis dass diese ein Mensch sind, auf die Seite zuzugreifen.
 
-Aufgrund der Risiken im Zusammenhang mit der Bereitstellung eines Wegs um die blockierung zu umgehen, ist es nicht empfohlen diese Methode bereitzustellen. Es ist Empfohlen, diese Methode nur zu aktivieren wenn Sie diese unbedingt benötigen. Dies könnten beispielsweise diese Situatuinen seien:
+Aufgrund der Risiken im Zusammenhang mit der Bereitstellung eines Wegs um die Blockierung zu umgehen, ist es nicht empfohlen diese Methode bereitzustellen. Es ist Empfohlen, diese Methode nur zu aktivieren wenn Sie diese unbedingt benötigen. Dies könnten beispielsweise diese Situatuinen seien:
```
eines Wegs => eines Weges

DanielRuf replied:
Situatuinen => Situationen
Es ist Empfohlen => Es ist empfohlen
seien => sein

DanielRuf commented on _docs/readme.de.md:
```diff
@@ -1092,7 +1095,7 @@ Module wurden zur Verfügung gestellt, um sicherzustellen, dass die folgenden Pa
 - [Werden Probleme auftreten, wenn ich CIDRAM gleichzeitig mit CDNs oder Caching-Diensten verwende?](#CDN_CACHING_PROBLEMS)
 - [Wird CIDRAM meine Website vor DDoS-Angriffen schützen?](#DDOS_ATTACKS)
-- [Wenn ich Module oder Signaturdateien über die Update-Seite aktiviere oder deaktiviere, sortiert siche diese alphanumerisch in der Konfiguration. Kann ich die Art der Sortierung ändern?](#CHANGE_COMPONENT_SORT_ORDER)
+- [Wenn ich Module oder Signaturdateien über die Update-Seite aktiviere oder deaktiviere, sortiert sie diese alphanumerisch in der Konfiguration. Kann ich die Art der Sortierung ändern?](#CHANGE_COMPONENT_SORT_ORDER)
```
sie => sich

DanielRuf commented on _docs/readme.de.md:
```diff
@@ -1245,7 +1248,7 @@ Erstens ist CIDRAM ein PHP-Paket, und arbeitet daher auf dem Rechner, auf dem PH
 Dies kann mit dedizierten Vor-Ort-Hardware-Lösungen, und/oder Cloud-basierte Lösungen wie dedizierten DDoS-Mitigationsdiensten, routing des DNS einer Domain über DDoS-resistente Netzwerke, Cloud-basierte Filterung, oder eine Kombination davon implementiert werden. Auf jeden Fall ist dieses Thema ein wenig zu komplex, um es mit nur ein oder zwei Paragraphen gründlich zu erklären, also würde ich em ...
 
-#### <a name="CHANGE_COMPONENT_SORT_ORDER"></a>Wenn ich Module oder Signaturdateien über die Update-Seite aktiviere oder deaktiviere, sortiert siche diese alphanumerisch in der Konfiguration. Kann ich die Art der Sortierung ändern?
+#### <a name="CHANGE_COMPONENT_SORT_ORDER"></a>Wenn ich Module oder Signaturdateien über die Update-Seite aktiviere oder deaktiviere, sortiert sie diese alphanumerisch in der Konfiguration. Kann ich die Art der Sortierung ändern?
```
sie => sich

Maikuolan committed:
Progress.

- Fixes some things that I'd forgotten to translate before.
- Implements new review items.
- Various other small related fixes.

Maikuolan commented on _docs/readme.de.md:
```diff
@@ -1079,75 +1125,79 @@ Obwohl diese doppelte Bedeutung von "CIDR" in einigen Fällen mehrdeutig werden
 #### <a name="WHAT_IS_A_FALSE_POSITIVE"></a>Was ist ein "Falsch-Positiv"?
 
-Der Begriff "Falsch-Positiv" (*Alternative: "falsch-positiv Fehler"; "falscher Alarm"*; Englisch: *false positive*; *false positive error*; *false alarm*), sehr einfach beschrieben, und in einem verallgemeinerten Kontext, verwendet wird, wenn eine Bedingung zu testen und wenn die Ergebnisse positiv sind, um die Ergebnisse dieser Tests zu entnehmen (dh, die Bedingung bestimmt wird positiv oder wah ...
+Der Begriff "Falsch-Positiv" (*Alternative: "falsch-positiv Fehler"; "falscher Alarm"*; Englisch: *false positive*; *false positive error*; *false alarm*), sehr einfach beschrieben, und in einem verallgemeinerten Kontext, wird verwendet, wenn eine Bedingung zu testen ist und wenn die Ergebnisse positiv sind (also wenn die Bedingung als positiv bewertet wird) jedoch die Bedingung als negativ erwar ...
```
@DanielRuf I'm not sure about this part (*"Der Begriff "Falsch-Positiv" ... identifiziert wurden"*). Spotted this part after merging, and I haven't touched it yet since the merge. Might need more work maybe.

DanielRuf replied:
It is also called "false positive" in German.
https://www.dict.cc/englisch-deutsch/false+positive.html
Or "Falschmeldung".

We can review it in the future.
